### PR TITLE
Deprecate io.{imshow,imshow_collection,show}

### PR DESCRIPTION
**Pull request by stefanv**
Partially addresses #7504

### CODE REVIEW

**Review by mkcor — APPROVED** [review]
Feels good!

**Comment by stefanv** [comment]
Raises warnings in the test suite, so would need to add those as expected before this will past.

**Review by lagru — APPROVED** [review]
Approving on the condition that my update fixed the CI.

File: skimage/io/tests/test_mpl_imshow.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def _reset_plugins():
     # Matplotlib 2.2.3 is the last supported version for python 2.7
     r"np.asscalar|\A\Z",
     r"The figure layout has changed to tight|\A\Z",
+    "is deprecated since version 0.25.* Please use .* to visualize images",
```
Comment (mkcor) [inline skimage/io/tests/test_mpl_imshow.py:36]: Good command of regular expressions! :grin:

Comment (mkcor) [reply skimage/io/tests/test_mpl_imshow.py:36]: Nice, I'm glad this is enough for CI to pass. I'll merge now then.

**Pull request merged**